### ISSUE TITLE
Unable to handle multi layered tile maps after upgrade to Godot 4.3

### ISSUE THREAD

**Issue by fjsventura** [AlexeyBond/godot-constraint-solving#27]
Hi,

Thank you for your great work. 
I was trying the upgrade for Godot 4.3 but I've been unable to sort it our even looking at the provided examples.

I have three layers in this examples and I believe I followed the proper configurations by including them in a WFC2DLayeredMap node:
![image](https://github.com/user-attachments/assets/e68dd32e-fa13-4ad5-b394-3a88a7a5afc4)

I believe I used the right object to configure the tileset in the generator (which is needed in my case, but I saw that the example provided didn't had this configured), for instance for the sample:
![image](https://github.com/user-attachments/assets/826227cd-b34d-42c9-867d-7b5397fc737f)

I can see that the configurations seem fine, but still when I try to run the application I fail this assertion:
````
func _ensure_tile_map_layer(map: Node) -> TileMapLayer:
	assert(map is TileMapLayer) # Here

	return map as TileMapLayer
```

Should I be using a different object type for the mapper in the ruler?

Thanks for your support

**Comment by AlexeyBond** [comment]
Hi. For multi-layer map (`WFC2DLayeredMap`) you should use [WFCLayeredMapMapper2D](https://github.com/AlexeyBond/godot-constraint-solving/blob/master/addons/wfc/problems/2d/mappers/mapper_2d_layered_map.gd) with `layer_mappers` specified for each layer.

Or just leave the field free, a correct mapper will be created by default.

**Comment by fjsventura** [comment]
Hi, 
Thanks for your quick answer. If I leave it empty I get an error here:
```
## Creates a mapper for given [param map] node.
## [br]
## Called when mapper is not provided in [member rules].
func _create_mapper(map: Node) -> WFCMapper2D:
	var mapper := mapper_factory.create_mapper_for(map)

	if mapper == null:
		var script: Script = map.get_script()
		push_error("Unsupported map node type: %s (%s)" % [
			map.get_class(),
			 "No script" if script == null else (
				script.get_global_name() if not script.get_global_name().is_empty() else script.resource_path
			)])
		@warning_ignore("assert_always_false")
		assert(false) # Here

	return mapper
```

It seems it is unable to find a valid implementation for the mapper. 

I was unable to move forward a bit with declaring them as you suggested:
![image](https://github.com/user-attachments/assets/c58988a8-debc-4439-b226-a21d3fab53f9)


Just had to remove the probabilities for some reason. 
Thank you very much.

**Comment by AlexeyBond** [comment]
Can you try replacing `"Node2D"` by `"Node"` [here](https://github.com/AlexeyBond/godot-constraint-solving/blob/master/addons/wfc/problems/2d/mappers/mapper_2d_factory.gd#L25) and try to run it without mapper configured?

Looks like I'm running into some code caching issues - on my machine it works. But `WFC2DLayeredMap` isn't a `Node2D` now, so it shouldn't.

**Comment by AlexeyBond** [comment]
I used `Node2D` as base class for `WFC2DLayeredMap` during developbent, and it was saved as Node2D with script in the [demo scene](https://github.com/AlexeyBond/godot-constraint-solving/blob/master/addons/wfc/examples/demo_wfc_2d_tilemaplayer_multilayer.tscn#L29-L31):

```
[node name="sample" type="Node2D" parent="."]
visible = false
script = ExtResource("3_orkfj")
mapper_factory = SubResource("Resource_xa0lu")
```

And then I changed the base class. But it didn't change in the demo. The demo still worked, so I forgot to change the check in mapper factory.

**Comment by fjsventura** [comment]
Interesting,

That way it works, yes. 

The only problems I'm facing right now is on my preconditions which may not be anything too serious I hope and for some reason in my code the WFC2DLayeredMap does not have a hide and show method. But with this the mapper was automatically created, yes.

**Comment by fjsventura** [comment]
It seems that the hide/show methods are only available in node2d, I guess...

**Comment by AlexeyBond** [comment]
The issue is fixed in version [1.6.1](https://github.com/AlexeyBond/godot-constraint-solving/releases/tag/1.6.1).

**Comment by fjsventura** [comment]
Thank you very much
